pac4j-jwt/src/main/java/org/pac4j/jwt/profile/JwtGenerator.java

```diff
@@ -0,0 +1,86 @@
+/*
+  Copyright 2012 - 2015 pac4j organization
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+ */
+package org.pac4j.jwt.profile;
+
+import com.nimbusds.jose.*;
+import com.nimbusds.jose.crypto.DirectEncrypter;
+import com.nimbusds.jose.crypto.MACSigner;
+import com.nimbusds.jwt.JWTClaimsSet;
+import com.nimbusds.jwt.SignedJWT;
+import org.pac4j.core.exception.TechnicalException;
+import org.pac4j.core.profile.UserProfile;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Date;
+
+/**
+ * Generates a JWT token from a user profile.
+ *
+ * @author Jerome Leleu
+ * @since 1.8.0
+ */
+public class JwtGenerator<U extends UserProfile> {
```

Comment:
Good question! This class is meant to generate a JWT from an existing user profile. The typical use case is an application where the user can login with Facebook and then generates a token to access web services. This was inspired by this discussion: https://groups.google.com/forum/?fromgroups#!topic/pac4j-users/kJMw8vvZBJY